metrics-datadog/src/main/scala/com/ovoenergy/effect/Datadog.scala

```diff
@@ -3,42 +3,90 @@ package com.ovoenergy.effect
 import java.net.InetSocketAddress
 
 import cats.effect._
+import com.ovoenergy.effect.Events.Event
 import com.ovoenergy.effect.Metrics.Metric
 import fs2.Chunk.array
 import fs2.io.udp._
 
 object Datadog {
 
+  // just for readability
+  type UTF8Bytes = Array[Byte]
+
+  private implicit class StringOps(s: String) {
+    def utf8Bytes: UTF8Bytes = s.getBytes("UTF-8")
+  }
+
+  private implicit class ByteArrayOps(bs: UTF8Bytes) {
+    def sep(char: Char)(other: UTF8Bytes): UTF8Bytes = (bs :+ char.toByte) ++ other
+    def colon: UTF8Bytes => UTF8Bytes = sep(':')
+    def pipe: UTF8Bytes => UTF8Bytes = sep('|')
+  }
+
   case class Config(
     agentHost: InetSocketAddress,
     metricPrefix: Option[String],
     globalTags: Map[String, String]
   )
 
+  /**
+   * Apparently the maximum UDP packet size is 65535 bytes (at the absolute maximum)
+   * and we have many different strings in a packet so we only allow each one to be 1Kb
```

Comment:
1000 chars is not 1kb. Even 1024 chars wouldn't be...

Reply:
oh yes I was getting mixed up :facepalm: